src/Workspaces/Remote/ServiceHub/Host/TestUtils.cs

```diff
@@ -192,7 +192,7 @@ public static async Task AppendAssetMapAsync(
             var compilationChecksums = await solution.CompilationState.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
             await compilationChecksums.FindAsync(solution.CompilationState, projectCone: null, AssetPath.FullLookupForTesting, Flatten(compilationChecksums), callback, map, cancellationToken).ConfigureAwait(false);
 
-            foreach (var frozenSourceGeneratedDocumentState in solution.CompilationState.FrozenSourceGeneratedDocumentStates?.States.Values ?? [])
+            foreach (var frozenSourceGeneratedDocumentState in solution.CompilationState.FrozenSourceGeneratedDocumentStates.States.Values ?? [])
```

Comment:
Is the `?? []` at the end still necessary? It's unclear from reading which thing here would be null, but I'm assuming it would have been a compiler warning if it was not possible to hit, so I'm a bit confused.

Reply:
you are right.  we don't need the ??[] here.